Always show the asset path in the workbench

mensinda opened the pull request:
This moves the resource button next to the locale. Additionally, the resource path of a text is now always shown and not hidden behind a tooltip:

![tmp](https://github.com/box/mojito/assets/3407462/c92a3896-ae64-45f4-aa38-b4b2ceb2a9d9)

aurambaj commented:
@mensinda  I like the concept in fact I always wanted to have the asset somehow directly shown. 

One problem though is that it is pretty common for it to be somewhat long which 
1) will make the UI cluttered with repetition 
2) overflow badly...  

for the overflow maybe we can restrict it to max size, truncating the start and just show the end of the path. Then when hovering on the text, we'd see the whole path.

for the cluttered part, I've been thinking to have a selector to choose different "view mode" next to the pagination. It would show more or less information based on the mode. 

One other info pretty useful to have is the "repository name" when doing cross repository searches.

So we could have a condensed view, which the current display and and expanded view which would show the repository name and repository path. To avoid having to deal with length of the path, I think it would make sense to just have the path on its own line.

Any thoughts on that? Wondering if you have other information besides the path that you feel should be directly accessible.

mensinda commented:
> for the overflow maybe we can restrict it to max size, truncating the start and just show the end of the path

Implemented by setting the max width to `256px` and `overflow: hidden`.

> for the cluttered part, I've been thinking to have a selector to choose different "view mode" next to the pagination

Implemented.

> One other info pretty useful to have is the "repository name" when doing cross repository searches.

I have also added this.

> Any thoughts on that? Wondering if you have other information besides the path that you feel should be directly accessible.

Not that I can currently think of.

aurambaj commented:
Thanks @mensinda ! Merging the PR as it but wondering about the use of color, ideally I'd avoid adding to many also it is not super obvious what the color conveys in that case. I'm thinking that may in that expended view it could be more explicit

Potentially, on its own line before the locale/string id, Have 2 labels prefixed with "repo" eg. "repo: demo" and other "asset: demo.properties". I think "grey" would be fine for those

aurambaj commented on webapp/src/main/resources/public/js/components/workbench/ViewModeDropdown.js:
```diff
@@ -0,0 +1,40 @@
+        return (
+            <DropdownButton  
+                id="WorkbenchStatusDropdown" 
+                title={this.props.intl.formatMessage({id: "search.viewMode.title"})}
```
Maybe update the dropdown title with the selected mode?

mensinda replied:
Done.

aurambaj commented on webapp/src/main/resources/public/js/components/workbench/TextUnit.js:
```diff
@@ -841,17 +868,66 @@ let TextUnit = createReactClass({
+        }
+
+        return (
+            <Label bsStyle='danger' bsSize='large' className="clickable" onClick={this.onRepositoryClick}>
```
I don't think we should be using the "danger" style here since it is just regular information. see the comment on the whole PR. Maybe having a prefix on the label to say "repository" or "repo" and put it on its own line?

mensinda replied:
Also, adding a new line is not that easy, since most of the space is already used. So adding a new line in the label, would make each text line even bigger.

Would you be fine with using the `secondary` color for the repository?

aurambaj replied:
btw, I just remembered that the asset info is already accessible by hovering on the text unit (and clicking on the icon drills down to search in that asset). I guess that's the approach i had to avoid cluttering everything at the time. 

we already have a few label, eg. "unused" can be added to the right of the locale.

I think we should keep the danger color for things that are actually problematic and not for regular display. 

In the full view, we can use more line for clarify, I think. Maybe something like. 

<img width="272" alt="Screenshot 2024-02-28 at 2 20 00 PM" src="https://github.com/box/mojito/assets/9702182/ee95e736-eba9-4e87-9fca-61025148c285">

mensinda replied:
Full:

![tmp](https://github.com/box/mojito/assets/3407462/91da2f93-b1f5-4e29-bb43-d8bf61ab602a)

Reduced:

![tmp](https://github.com/box/mojito/assets/3407462/252b4aa7-e7fa-4e05-9d6e-42e198d8af48)

aurambaj commented on webapp/src/main/resources/public/js/components/workbench/TextUnit.js:
```diff
@@ -841,17 +868,66 @@ let TextUnit = createReactClass({
+
+        return (
+            <OverlayTrigger placement="top" overlay={assetPathTooltip}>
+                <Label style={{maxWidth: "256px", overflow: "hidden"}} bsStyle='info' bsSize='large' className="clickable" onClick={this.onAssetPathClick}>
```
same comment on the style/color and the prefix/be on its own line.

aurambaj commented on webapp/src/main/resources/properties/en.properties:
```diff
@@ -222,6 +222,12 @@ search.result.empty=There are no results. Update your search parameter to find w
+search.viewMode.title=View mode
+
+search.viewMode.FULL=Full
+search.viewMode.REDUCED=Reduced
```
Reduced -> Standard?

mensinda replied:
Maybe I just add a 4th standard state (and make it the default) which has the same information as the current layout?

mensinda replied:
This is now implemented

aurambaj commented on webapp/src/main/resources/public/js/stores/workbench/ViewModeStore.js:
```diff
@@ -0,0 +1,28 @@
+import alt from "../../alt";
+import ViewModeActions from "../../actions/workbench/ViewModeActions";
+import { readCookie, writeCookie } from "../../utils/Cookie";
```
can we store this in local storage instead since it is client side only configuration?

mensinda replied:
Do you mean with `local storage` inside the DB in the server?

If you mean just saved in JS (it is gone if you refresh the page), then I would like to keep it a little more persistent so that I don't have to change it back every time.

aurambaj replied:
browser [local storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage), that should support browser restart, etc

mensinda replied:
Ah, I wasn't aware of that. I have changed it to the local storage now.

aurambaj commented on webapp/src/main/resources/sass/mojito.scss:
```diff
@@ -462,15 +462,51 @@ div.textunit label {
+
+    grid-template-columns: min-content min-content min-content fit-content(100%) 1fr min-content min-content;
+    grid-template-rows: min-content 1fr min-content min-content 1fr;
+    grid-template-areas:
```
nice
